File: codemap/indexer/testdata/php/src/Orders/OrderService.php
<?php

namespace Acme\Orders;

use Acme\Store\Repo;
use Acme\Store\PgRepo;
use function Acme\Util\clean;

/** Places orders. */
class OrderService
{
    private Repo $repo;

    public function __construct(?Repo $repo = null)
    {
        $this->repo = $repo ?? new PgRepo();
    }

    public function place(Order $order): int
    {
        if (!$this->validate($order)) {
            return 0;
        }
        clean($order);
        return $this->repo->save($order) + strlen(PgRepo::table());
    }

    private function validate(Order $order): bool
    {
        return $order->id > 0 && self::limit() > 0;
    }

    private static function limit(): int
    {
        return 10;
    }
}

File: codemap/indexer/testdata/dart/lib/src/order_service.dart
import 'package:shop/src/store/repo.dart';
import 'util.dart' as util;

/// Places orders.
class OrderService {
  final Repo repo;

  OrderService(this.repo);

  factory OrderService.pg() => OrderService(PgRepo());

  int place(Order order) {
    if (!_validate(order)) {
      return 0;
    }
    util.clean(order);
    return repo.save(order) + PgRepo.table.length;
  }

  bool _validate(Order order) => order.id > 0;
}

class Order {
  final int id;
  const Order(this.id);
}

int run() => OrderService.pg().place(const Order(1));
